appservice/src/createAppService/SiteNameStep.ts

```diff
@@ -122,12 +134,101 @@ export class SiteNameStep extends AzureNameStep<SiteNameStepWizardContext> {
+            throw new Error(vscode.l10n.t('Internal Error: A location is required when validating a site name with regional CNA.'));
+        } else if (domainNameScope === DomainNameLabelScope.ResourceGroup && !resourceGroupName) {
+            throw new Error(vscode.l10n.t('Internal Error: A resource group name is required for validating this level of domain name scope.'));
+        }
+
+        const apiVersion: string = '2024-04-01';
+        const location: AzExtLocation = await LocationListStep.getLocation(context);
+        const authToken: string = nonNullValueAndProp((await context.credentials.getToken() as { token?: string }), 'token');
+
+        // Todo: Can replace with call using SDK once the update is available
+        const options: AzExtRequestPrepareOptions = {
+            url: `${context.environment.resourceManagerEndpointUrl}subscriptions/${context.subscriptionId}/providers/Microsoft.Web/locations/${location.name}/checknameavailability?api-version=${apiVersion}`,
+            method: 'POST',
+            headers: createHttpHeaders({
+                'Content-Type': 'application/json',
+                'Authorization': `Bearer ${authToken}`,
+            }),
+            body: JSON.stringify({
+                name: siteName,
+                type: 'Site',
+                autoGeneratedDomainNameLabelScope: domainNameScope,
+                resourceGroupName: domainNameScope === DomainNameLabelScope.ResourceGroup ? resourceGroupName : undefined,
+            }),
+        };
+
+        const client = await createGenericClient(context, undefined);
+        const pipelineResponse = await client.sendRequest(createPipelineRequest(options)) as AzExtPipelineResponse;
+        const checkNameResponse = pipelineResponse.parsedBody as {
+            hostName?: string;
+            message?: string;
+            nameAvailable?: boolean;
+            reason?: string;
+        };
+
+        if (!checkNameResponse.nameAvailable) {
+            // If site name input is greater than 46 chars, ignore result of regional CNA because it inherently has a shorter character limit than Global CNA
+            if (domainNameScope === DomainNameLabelScope.Global && siteName.length > regionalCNAMaxLength) {
+                // Ensure the error message is the expected character validation error message before ignoring it
+                if (checkNameResponse.message && /must be less than \d{2} chars/i.test(checkNameResponse.message)) {
+                    return undefined;
+                }
+            }
+            return checkNameResponse.message;
+        }
+
+        return undefined;
+    }
+
+    private async asyncValidateUniqueARMId(context: SiteNameStepWizardContext, client: WebSiteManagementClient, siteName: string, resourceGroupName?: string): Promise<string | undefined> {
+        if (!resourceGroupName) {
+            context.relatedNameTask = this.generateRelatedName(context, siteName, this.getRelatedResourceNamingRules(context));
```

Comment:
Should you be overwriting the relatedNameTask like this? What if one already exists?

Reply:
Good point, I should probably use `??=` here